pkg/dynamicrp/backend/dynamicresource_controller.go

```diff
@@ -64,17 +76,57 @@ func (c *DynamicResourceController) selectController(request *ctrl.Request) (ctr
 		return nil, fmt.Errorf("invalid resource ID: %q", request.ResourceID)
 	}
 
+	resourceType, err := c.fetchResourceType(ctx, id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch resource type: %w", err)
+	}
+
 	options := ctrl.Options{
 		DatabaseClient: c.DatabaseClient(),
 		ResourceType:   id.Type(),
 	}
 
 	switch ot.Method {
 	case v1.OperationDelete:
+		if hasCapability(resourceType, datamodel.CapabilitySupportsRecipes) {
+			return NewRecipeDeleteController(options, c.engine, c.configurationLoader)
+		}
+
 		return NewInertDeleteController(options)
+
 	case v1.OperationPut:
+		if hasCapability(resourceType, datamodel.CapabilitySupportsRecipes) {
+			return NewRecipePutController(options, c.engine, c.configurationLoader)
+		}
+
 		return NewInertPutController(options)
+
 	default:
 		return nil, fmt.Errorf("unsupported operation type: %q", request.OperationType)
 	}
```

Comment:
Just curious - where is the controller for GET requests for dynamic rp resources?